dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java

```diff
@@ -771,13 +772,15 @@ private T createExtension(String name, boolean wrap) {
                 }
 
                 if (CollectionUtils.isNotEmpty(wrapperClassesList)) {
-                    for (Class<?> wrapperClass : wrapperClassesList) {
-                        Wrapper wrapper = wrapperClass.getAnnotation(Wrapper.class);
-                        if (wrapper == null
-                            || (ArrayUtils.contains(wrapper.matches(), name) && !ArrayUtils.contains(wrapper.mismatches(), name))) {
-                            instance = injectExtension((T) wrapperClass.getConstructor(type).newInstance(instance));
-                            instance = postProcessAfterInitialization(instance, name);
-                        }
+                    wrapperClassesList = wrapperClassesList.stream().filter(wrapperClz -> {
+                        Wrapper wrapper = wrapperClz.getAnnotation(Wrapper.class);
+                        return (wrapper == null) ||
+                            ((ArrayUtils.isEmpty(wrapper.matches()) || ArrayUtils.contains(wrapper.matches(), name)) &&
+                                !ArrayUtils.contains(wrapper.mismatches(), name));
+                    }).collect(Collectors.toList());
+                    for (Class<?> wrapperClz : wrapperClassesList) {
+                        instance = injectExtension((T) wrapperClz.getConstructor(type).newInstance(instance));
+                        instance = postProcessAfterInitialization(instance, name);
```

Comment:
I think streaming is not a good option used here. Normal for iterating can be more memory efficient while clear enough.

Reply:
got it